### PR TITLE
Validate license headers in YAML files

### PR DESCRIPTION
**Pull request by antoineco**
On top of validating Go files, validate our YAML manifests.

### CODE REVIEW

File: .github/workflows/config/reviewdog.yaml [hunk 1]
```diff
@@ -12,3 +12,12 @@ runner:
     errorformat:
       - '%A%f:%l: %m'
       - '%C%.%#'
+  license-yaml:
+    cmd: >-
+      boilerplate-check check
+      --boilerplate=hack/boilerplate/boilerplate.yaml.txt
+      --file-extension=yaml
+      --exclude='(^(\.|config/samples/))'
```
Comment (antoineco) [inline .github/workflows/config/reviewdog.yaml:20]: I wasn't sure whether we wanted to enforce headers inside samples or not.
Thoughts?

Comment (JeffNeff) [reply .github/workflows/config/reviewdog.yaml:20]: +1 they should be there

Comment (odacremolbap) [reply .github/workflows/config/reviewdog.yaml:20]: No strong opinion but mostly concur with Jeff.

File: config/300-filter-crd.yaml [hunk 2]
```diff
@@ -4,7 +4,7 @@
 # you may not use this file except in compliance with the License.
 # You may obtain a copy of the License at
 #
-#     https://www.apache.org/licenses/LICENSE-2.0
+#     http://www.apache.org/licenses/LICENSE-2.0
```
Comment (cab105) [inline config/300-filter-crd.yaml:7]: Does it matter if using http vs. https, or did the linter complain?

Comment (antoineco) [reply config/300-filter-crd.yaml:7]: It complained yes. The only thing it ignores is the year.

I don't think it matters personally, but I won't why these 4 specific files had HTTPS.

Comment (cab105) [reply config/300-filter-crd.yaml:7]: Guess it depends on what was used as the basis for the copy/pasta boilerplate.

File: .github/workflows/config/reviewdog.yaml (hunk 1 above)
Comment (antoineco) [reply .github/workflows/config/reviewdog.yaml]: Thanks for the comments!
I added missing headers to all samples reported by the linter.

**Review by odacremolbap — APPROVED** [review]

**Pull request merged**